core/vision/ar_tag_detector.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+from typing import List
+from algorithms.AR_tag import Tag
+import core
+import algorithms
+
+# Dict to hold the obstacle info
+ar_tag = {}
+
+
+async def async_ar_tag_detector():
+    """
+    Async function to find obstacles.
+    """
+    while True:
+        reg_img = core.vision.camera_handler.grab_regular()
+
+        tags, reg_img = algorithms.AR_tag.detect_ar_tag(reg_img)
+
+        core.vision.feed_handler.handle_frame("artag", reg_img)
+
+        if len(tags) > 0:
+            ar_tag["detected"] = True
+            ar_tag["tags"] = tags
+        else:
+            ar_tag["detected"] = False
+            ar_tag["tags"] = []
+
+        await asyncio.sleep(1 / 30)
```

Comment:
Explain why this sleep is here and the duration - I assume it is also 1/30 to make it the same as the sleep of the camera feeds and not a pure coincidence - maybe this framerate sleep parameter should be a global constant so that you don't have to hunt through the codebase to change framerate.

Reply:
Yeah, will address this in future PR along with other ZED constants. The sleep is here so python's asynchronous state machine can schedule another task while we are waiting for new frames.